[FEAT] Join 페이지 레이아웃 구축

Da-Yeon-Kim opened the pull request:
## 주요 변경사항 ✔️
Join 페이지의 컴포넌트를 분리하고, 레이아웃만 맞추었습니다.
Calendar는 일단 건규 씨가 작성해둔 코드 복붙해 두었어요😅
아직 선호/비선호 음식 컴포넌트는 안 만들어져 있어서 자리만 잡아두었습니다.
덤으로 이전에 만든 Create 페이지도 공용 Container를 사용해 레이아웃을 잡도록 수정했어요!

## 리뷰어에게... 🙏
1. `Container`에 padding이 없으니 작은 화면에서 너무 양 끝에 딱 붙길래 추가했습니다.
2. 캘린더가 폰트가 Pretendard가 아니라서 그 부분을 추가했습니다. 저희 폰트를 Pretendard만 사용할 것 같은데, 아예 Common CSS를 만들어서 전체 적용을 해버리면 좋을 것 같아요. 보아하니 폰트만 다른 게 아니고, 각자 페이지마다 글자 크기나 굵기도 다르더라고요. 통일의 필요성이 있어 보입니다.

시험이 금요일에 끝나서😥 일단 이거라도 미리 PR 보내놓습니다!

+) package.json에 msw인가? 그게 두 줄 중복되어 있는 것 같은데 확인 부탁드려요!

## 관련 이슈

closes #61

Da-Yeon-Kim committed:
feat(Join): Join 페이지 레이아웃 구축

- JoinTitle, JoinBtn, JoinCalendar, JoinFood로 섹션 분리
- JoinCalendar는 DefalutCalendar 가져와 표시
- JoinFood는 추후 공용 컴포넌트 생성 시, 가져올 예정

- 공용 컴포넌트인 Container에 padding 추가 (sm 크기의 padding은 따로)
- DefalutCalendar의 폰트를 Pretendard로 수정

 #61

Da-Yeon-Kim committed:
feat(Create): Create 페이지도 공용 Container 사용

 #61

geongyu09 commented on src/components/features/Join/JoinTitle/index.tsx:
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+export const JoinTitle: React.FC = () => {
+  const title = '철수 모임';
```
해당 값은 아마 서버에서 받아와야 할 것 같은데, 찾아보니 해당 api가 없네요..! 
백엔드와 함께 이야기 해봐야 할 것 같아요!

Da-Yeon-Kim replied:
그렇네요! 찾아보니 모임의 이름 정보를 가져오는 API가 없네요... 다음 주 회의 때 말해볼게요!

geongyu09 commented on src/pages/Join/index.tsx:
```diff
@@ -1,3 +1,16 @@
+      <JoinTitle />
+      <JoinCalendar />
+      <JoinFood />
+      <JoinBtn />
```
ui의 책임별로 잘 나누어진 것 같아 정말 좋네요! 
다만 개인적으로 ui를 재사용하기에는 완전히 적절한 코드는 아닌 것 같다는 생각이 들어요..!

join, edit 페이지는 다른 데이터 및 데이터를 가져오는 로직만 다르고 동일한 ui로 보여질 것 같은데, 이를 잘 수행할 수 있도록 컴포넌트 분리에 대해서 고민해봐도 좋아보입니다! 

간단히 저의 생각을 말하면 title, calendar, food, btn이 거희 완전히 동일하게 레이아웃까지 다시 재사용 될 수 있을테니 ui 코드부분만 구현을 해주고, 모든 외부 로직들(여기서는 데이터를 받아오는 로직, 백엔드로 요청 보내는 로직 등)을 전부 상위에서 구현 후에 주입해주는 방식으로 구성하는 것이 좋아보입니다!

물론 아직 완성된 코드가 아니기에, 저보다 더 좋은 생각을 가지고 있을 수도 있을 것 같다는 생각이 들지만! 
앞으로 개발하면서 참고용으로만 읽어주시면 감사할 것 같습니다!

geongyu09 commented on src/components/common/layouts/Container/index.tsx:
```diff
@@ -44,10 +44,15 @@ const Inner = styled.div<
   justify-content: ${({ justifyContent }) => justifyContent ?? 'left'};
   align-items: ${({ alignItems }) => alignItems ?? 'flex-start'};
   gap: ${({ gap }) => gap ?? '0'};
+  padding: 20px;
```
혹시 해당 컴포넌트를 사용하는 다른 곳에서는 추가되거나 변경된 값에 대해서 영향을 안받나요??

Da-Yeon-Kim replied:
영향... 받습니다! 그래서 확인해 보면 Home, Create 페이지도 패딩이 생겼어요.
공통적으로 추가해보는 게 어떤지 여쭤보려고 PR에 변경 사항 포함해둔 거예요...ㅎㅎ
개인적으로는 패딩이 있는 편이 화면이 줄었을 때도 보기가 좋아 보이는데, 의견을 물어보고 싶어요!

geongyu09 replied:
음.. 아무래도 공통 컴포넌트인데 고정적인 값이 존재해버리면 그만큼 재사용성이 떨어질 수 있을 것 같아요!
개인적으로는 이런 값들을 props로 받아서 사용할 수 있도록 하면 더 좋아보이네요!

Da-Yeon-Kim replied:
조언해주신 내용에 동의해서 props로 받아서 사용하도록 수정했습니다😊

geongyu09 reviewed (approved):
시험은 잘 보셨나요..? 저도 오늘 시험이 끝나서 리뷰를 조금 늦게 달게 되었네요..🥲

일단 캘린더의 폰트를 주지 않은 것은 언급하신대로 모든 폰트가 pretendard일텐데 각각에 주는 것은 올바르지 않다고 생각해서 제가 했던 모든 작업에서 폰트 설정은 빠져있습니다
전체적으로 폰트를 수정하는 것은 어렵지 않으니, 해당 폰트 작업은 나중에 작업 하는게 좋아보입니다!
이 작업을 위해서 지금 굳이 지역적으로 폰트를 작성하지 않아도 될 것 같습니다!

package.json에 중복된것은 현재는 큰 문제가 되지 않을겁니다! 
그냥 한 줄 지우면 되는데, 이것도 따로 이슈 파서 작업 하는게 좋아보이네요 (충돌 위험..)
나중에 이것도 수정해봅시다

수고많으셨어요!

HyeonChae1104 reviewed (approved):
시험기간인데 수고 많으셨습니다!!
다연님 늘 코드도 깔끔하게 잘짜시네요!!
Join 페이지 레이아웃도 체계적으로 잡아주신 것 같아요!!

음식 선호/비선호 부분은 제가 마이페이지에서 얼른 작업해보도록 하겠습니다.!!

Da-Yeon-Kim committed:
feat(Container): padding props로 받아서 사용

- 기존에는 지정되던 padding을 각자 props로 받아서 사용할 수 있도록 수정
- 기본값은 20px
- 원래 화면이 작아지면 padding이 줄어들게 했는데, 일단 이 기능은 보류
